lib/code_corps/policy/user_invite.ex

```diff
@@ -0,0 +1,42 @@
+defmodule CodeCorps.Policy.UserInvite do
+  @moduledoc ~S"""
+  Handles `CodeCorps.User` authorization of actions on `CodeCorps.UserInvite`
+  records.
+  """
+
+  alias CodeCorps.{Policy.Helpers, User}
+
+  @doc ~S"""
+  Returns true if the specified `CodeCorps.User` is allowed to create a
+  `CodeCorps.UserInvite` using the specified attributes.
+  """
+  @spec create?(User.t(), map) :: boolean
+  def create?(
+        %User{id: user_id} = user,
+        %{"project_id" => _, "inviter_id" => inviter_id} = params
+      )
+      when user_id == inviter_id do
+    user_role =
+      params
+      |> Helpers.get_project()
+      |> Helpers.get_membership(user)
+      |> Helpers.get_role()
+
+    new_role = Map.get(params, "role")
+    do_create?(user_role, new_role)
+  end
```

Comment:
Two conditions for the case of a project invite

- user is the inviter
- user has a role with authority to invite a user with the specified role